src/main/java/com/github/libgraviton/messaging/strategy/jms/JmsConnection.java

```diff
@@ -159,4 +148,56 @@ protected void closeConnection() throws CannotCloseConnection {
             queue = null;
         }
     }
+
+    public static class Builder<JmsBuilder extends Builder> extends QueueConnection.Builder<JmsBuilder> {
+
+        protected ConnectionFactory connectionFactory;
+
+        protected String messageSelector;
+
+        /**
+         * Apply defaults
+         */
+        public Builder() {
+            port(61616);
+        }
+
+        /**
+         * Defines the JMS connection factory.
+         *
+         * @param connectionFactory The connection factory
+         *
+         * @return self
+         */
+        public JmsBuilder connectionFactory(ConnectionFactory connectionFactory) {
+            this.connectionFactory = connectionFactory;
+            return (JmsBuilder) this;
```

Comment:
The only solution I found is this one: http://stackoverflow.com/questions/17164375/subclassing-a-java-builder-class/34741836#34741836. I wasn't really happy with this either. 
What do you vote for, `getThis()` or unchecked casts?